### PR TITLE
feat(compass-query-bar): selectively autocomplete query history in additional options COMPASS-8118

### PR DESCRIPTION
**Pull request by VivianTNT**
## Description

Filter bar is unchanged and will display all saved queries. The additional option editors (i.e. project, sort, collation, etc.) will only display saved queries that contain that specific property. 

image

image

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-query-bar/src/components/option-editor.tsx [hunk 1]
```diff
@@ -83,6 +83,7 @@ const insightsBadgeStyles = css({
 });
 
 type OptionEditorProps = {
+  barName: string;
```
Comment (Anemy) [inline packages/compass-query-bar/src/components/option-editor.tsx:86]: How do you feel about renaming this property to `optionName`? We don't refer to these as editors as bars (only the whole query bar) so I'm thinking we can make it a bit more descriptive.

Comment (VivianTNT) [reply packages/compass-query-bar/src/components/option-editor.tsx]: Agreed, just changed it

**Review by Anemy — APPROVED** [review]

File: packages/compass-query-bar/src/components/option-editor.tsx [hunk 2]
```diff
@@ -158,7 +160,14 @@ export const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
     return isQueryHistoryAutocompleteEnabled
       ? createQueryWithHistoryAutocompleter(
           savedQueries
-            .filter((query) => !('update' in query))
+            .filter((query) => {
+              const isOptionNameInQuery =
+                optionName !== 'filter'
+                  ? optionName in query.queryProperties
+                  : true;
```
Comment (Anemy) [inline packages/compass-query-bar/src/components/option-editor.tsx:167]: Another way to write this is
```suggestion
              const isOptionNameInQuery = optionName === 'filter' || optionName in query.queryProperties;
```

Comment (Anemy) [reply packages/compass-query-bar/src/components/option-editor.tsx]: Ah oops, this probably makes the reformat check fail, need to run reformat on it.

**Review by Anemy — APPROVED** [review]

**Pull request merged**